### DISCUSSION TITLE
tup也是个用lua写脚本的构建系统，或许可以从他那边学到什么

### DISCUSSION THREAD

**Discussion by hh-wu** [xmake-io/xmake#1561] [General]
> 注：如果是`问题报障`或者`特性请求`，请选择`报告缺陷`，`特性请求`模板，否则一律不回复。

### 描述讨论详情
最近在看odrive，发现开发者用了一个叫tup的构建系统，编译firmware的速度非常快。或许我们可以研究一下tup的原理，加入到xmake中。
tup的脚本看起来令人头疼。。。

tup主页是：http://gittup.org/tup/index.html

它里面介绍了一个基于文件系统的构建方案，看上去还挺吸引人的

**Comment by waruqi** [comment]
不知道你要表达什么，xmake 构建速度也很快，基本跟 ninja 持平，有基准测试
